spec/lib/api_entity_spec.rb

```diff
@@ -17,6 +17,22 @@ def self.name
     end
   end
 
+  describe '#resource_type' do
+    subject(:resource_type) { mock_entity.new(attributes).resource_type }
+
+    context 'when initialized with the resource type' do
+      let(:attributes) { { resource_type: 'foo' } }
+
+      it { is_expected.to eq('foo') }
+    end
+
+    context 'when initialized without the resource type' do
+      let(:attributes) { {} }
+
+      it { is_expected.to eq('mock_entity') }
```

Comment:
this is weird 🤔

Reply:
How comes? Have you looked at the mock_entity implementation? It might help beyond just this diff

Reply:
https://github.com/trade-tariff/trade-tariff-frontend/blob/add_resource_type/spec/lib/api_entity_spec.rb#L14-L16